contrib/cirrus/lib.sh

```diff
@@ -151,16 +165,118 @@ install_ooe() {
 }
 
 showrun() {
-    if [[ "$1" == "--background" ]]
+    local context
+    context=($(caller 0))
+    echo '+ '$(printf " %q" "$@")"  # ${context[2]}:${context[0]} in ${context[1]}()" > /dev/stderr
+    "$@"
+}
+
+comment_out_storage_mountopt() {
+    local FILEPATH=/etc/containers/storage.conf
+    echo ">>>>>"
+    echo ">>>>> Warning: comment_out_storage_mountopt() is modifying $FILEPATH"
+    echo ">>>>>"
+    sed -i -r -e 's/^(mountopt = .+)/#\1/' $FILEPATH
+}
+
+in_podman() {
+    req_env_var IN_PODMAN_NAME GOSRC HOME OS_RELEASE_ID
+    [[ -n "$@" ]] || \
+        die 7 "Must specify FQIN and command with arguments to execute"
+    local envargs
+    local envname
+    local envvalue
+    local envrx='(^CIRRUS_.+)|(^BUILDAH_+)|(^STOTAGE_)|(^CI$)|(^CROSS_TARGET$)|(^IN_PODMAN_.+)'
+    for envname in $(awk 'BEGIN{for(v in ENVIRON) print v}' | \
+                     egrep "$envrx" | \
+                     egrep -v "CIRRUS_.+_MESSAGE" | \
+                     egrep -v "$SECRET_ENV_RE")
+    do
+        envvalue="${!envname}"
+        [[ -z "$envname" ]] || [[ -z "$envvalue" ]] || \
+            envargs="${envargs:+$envargs }-e $envname=$envvalue"
+    done
+    # Back in the days of testing under PAPR, containers were run with super-privledges.
+    # That behavior is preserved here with a few updates for modern podman behaviors.
+    # The only other additions/changes are passthrough of CI-related env. vars ($envargs),
+    # some path related updates, and mounting cgroups RW instead of the RO default.
+    showrun podman run -i --name $IN_PODMAN_NAME \
+                   $envargs \
+                   --net=host \
+                   --net="container:registry" \
+                   --security-opt label=disable \
+                   --security-opt seccomp=unconfined \
+                   --cap-add=all \
+                   -e "GOPATH=$GOPATH" \
+                   -e "IN_PODMAN=false" \
+                   -e "DIST=$OS_RELEASE_ID" \
+                   -e "CGROUP_MANAGER=cgroupfs" \
+                   -v "$GOSRC:$GOSRC:z" \
+                   --workdir "$GOSRC" \
+                   -v "$HOME/auth:$HOME/auth:ro" \
+                   -v /sys/fs/cgroup:/sys/fs/cgroup:rw \
+                   -v /dev/fuse:/dev/fuse:rw \
+                   $@
```

Comment:
Careful! `$@` is magic, and must always be enclosed in double quotes: `"$@"`

Reply:
IIRC, it was giving me trouble when I did that but perhaps for some other reason.  I'm going to leave this one as-is since this function is so critical.  I'll come back and try quoting it if there's time and the thing runs fine after all the other changes.

Reply:
okay it passed, trying with the quotes...